### PR TITLE
[RB-13] Add Footer 

### PR DESCRIPTION
**Pull request by brandon91warren**
## Description
Added Footer to pages
## Changes
Added Footer to pages
## Screenshots
Screenshot 2024-11-21 at 9 41 38 AM
Screenshot 2024-11-21 at 9 41 01 AM

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import Router from './routes/Router';
 import { AuthProvider } from './context/AuthProvider';
 import MainLayout from './layouts/MainLayout';
+import Footer from "./layouts/Footer";
```
Comment (in43sh) [inline src/App.jsx]: If you look at the file `src\layouts\MainLayout.jsx` or (`src/layouts/MainLayout.jsx` if you are on Mac) you'll see that the footer is already imported and used there so there is no need for you to import it in `App.jsx`. Also, I'm pretty sure that we have a linter that doesn't allow it that you only import a component or create a variable and never use it. So, you can remove Footer from `App.jsx`.

File: src/assets/images/Footer.jpg
Comment (in43sh) [inline src/assets/images/Footer.jpg:1]: You should try to implement this as HTML markup in your `Footer.jsx` component file using Tailwind classes. There is no need to use an image.

File: src/layouts/Footer.jsx [hunk 2]
```diff
@@ -1,7 +1,16 @@
 import React from 'react';
+import footerImage from '../assets/images/footer.jpg'; //
 
 const Footer = () => {
-  return <></>;
+  return (
+      <footer style={{ width: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
```
Comment (in43sh) [inline src/layouts/Footer.jsx]: You should be using Tailwind classes here. We want to try to avoid using inline styling at all the whole project.

**Review by in43sh — APPROVED** [review]

File: src/assets/images/Footer.jpg
Comment (in43sh) [reply src/assets/images/Footer.jpg:1]: Also, implementing the footer this way makes our footer not responsive. You can see it if you change the browser width locally.

**Review by in43sh — CHANGES_REQUESTED** [review]
Sorry, I approved this PR by mistake. Also, there is an error in Netlify preventing it from successfully creating a build. Until there are no errors, it will not deploy our app on the frontend.

File: src/App.jsx (hunk 1 above)
Comment (brandon91warren) [reply src/App.jsx]: Thanks. Updated and reolved

File: src/assets/images/Footer.jpg
Comment (brandon91warren) [reply src/assets/images/Footer.jpg:1]: Thanks, will update in the next pull request

File: src/layouts/Footer.jsx (hunk 2 above)
Comment (brandon91warren) [reply src/layouts/Footer.jsx]: Will update in next pull request

File: src/layouts/Footer.jsx [hunk 3]
```diff
@@ -1,7 +1,15 @@
 import React from 'react';
```
Comment (in43sh) [inline src/layouts/Footer.jsx]: Importing React here is not required so you can remove it.

File: src/layouts/Footer.jsx [hunk 4]
```diff
@@ -1,7 +1,15 @@
...
 
 const Footer = () => {
-  return <></>;
+  return (
+    <footer className="bg-gray-800 py-6 text-black">
```
Comment (in43sh) [inline src/layouts/Footer.jsx]: You don't need the `<div>` container here. In general, it's a good practice to use as few `<div>` elements as possible as it's not good for SEO and the less elements you have the easier it is to read the code.

Comment (in43sh) [reply src/layouts/Footer.jsx]: Also, it's good to keep all related classes together. For example, I would move all text related classes to `<p>` tag. That would be:
```
<p className="text-sm text-black">
```

Comment (in43sh) [reply src/layouts/Footer.jsx]: You could change your html markup to this:
```
<footer className="bg-gray-800 flex justify-center py-6">
  <p className="text-sm text-black">
    Created and developed by Team #6 for the CTD Practicum
  </p>
</footer>
```

**Review by in43sh — CHANGES_REQUESTED** [review]

File: src/layouts/Footer.jsx (hunk 3 above)
Comment (EvgeniiRychkov) [reply src/layouts/Footer.jsx]: Linter requires it
error  'React' must be in scope when using JSX

**Review by brandon91warren — COMMENTED** [review]
thanks for the feedback. Updates will be in next pull request

File: src/layouts/Footer.jsx (hunk 3 above)
Comment (brandon91warren) [reply src/layouts/Footer.jsx]: Evgenii is correct. Although I was able to bypass this by turning the rule off in eslint.json. Please advise if correct 
Screenshot 2024-11-21 at 2 22 44 PM

File: tailwind.config.js [hunk 5]
```diff
@@ -17,6 +17,7 @@ export default {
         yellow: '#FFB347', // used for buttons, text, and border in outline buttons
         darkGreen: '#007185', // used for buttons, text, and border in outline buttons
         lightBlue: '#F5F9FA', // used for backgrounds of book cards, footer, etc.
+        footerBg: '#F5F9FA', // alias for the lightBlue color, specifically for the footer
```
Comment (in43sh) [inline tailwind.config.js]: There is already a color `#F5F9FA`. It's called `lightBlue` in the same config file. You don't need to create a new footer color.

**Review by in43sh — CHANGES_REQUESTED** [review]

**Review by in43sh — APPROVED** [review]

**Review by liuba83 — APPROVED** [review]

**Pull request merged**